tests/test_appclass.py

```diff
@@ -39,3 +43,44 @@ def test_dependencies(generate_app):
     with pytest.raises(traitlets.TraitError):
         app.version_to_install = "v22.11.0"
     app.version_to_install = "v22.11.1"
+
+
+def test_app_watch(tmp_path):
+    """Test the aiidalab app watch responsive to the app path changes."""
+
+    @dataclass
+    class DummyApp:
+        path: Path
+        x: int = 0
+        y: int = 0
```

Comment:
Unused variable
```suggestion
```

Also the method `refresh` is not used.

Reply:
Yes, I remove `y` but I think it is better to keep `refresh` method. It is not hinted by this test but it is inside the thread target function `check_path_exists_changed`.